azkaban-web-server/src/main/java/azkaban/flowtrigger/database/JdbcFlowTriggerInstanceLoaderImpl.java

```diff
@@ -95,16 +95,18 @@
               Status.SUCCEEDED.ordinal(),
               Constants.UNASSIGNED_EXEC_ID);
 
-  private static final String SELECT_RECENTLY_FINISHED =
+  private static final String SELECT_RECENTLY_FINISHED = String.format(
       "SELECT execution_dependencies.trigger_instance_id,dep_name,starttime,endtime,dep_status,"
           + "cancelleation_cause,project_id,project_version,flow_id,flow_version,"
           + "project_json, flow_exec_id \n"
           + "FROM execution_dependencies JOIN (\n"
           + "SELECT trigger_instance_id FROM execution_dependencies WHERE trigger_instance_id not in (\n"
```

Comment:
Why don't you just select trigger_instance_id where dep_status = SUCCEEDED or CANCELLED here? Will the sql query be simplified?

Reply:
yeah, good catch